src/main/java/org/jenkinsci/plugins/envinject/EnvInjectPrebuildJobProperty.java

```diff
@@ -0,0 +1,390 @@
+                    build.setResult(Result.FAILURE);
+                    throw new Run.RunnerAbortedException();
+                }
+
+                //Evaluate Groovy script
+                Map<String, String> groovyMapEnvVars = envInjectEnvVarsService.executeAndGetMapGroovyScript(logger, info.getGroovyScriptContent(), infraEnvVarsNode);
+
+                final Map<String, String> propertiesVariables = envInjectEnvVarsService.getEnvVarsPropertiesJobProperty(rootPath,
+                        logger, info.isLoadFilesFromMaster(),
+                        info.getPropertiesFilePath(), info.getPropertiesContentMap(previousEnvVars),
+                        infraEnvVarsMaster, infraEnvVarsNode);
+
+                //Get variables get by contribution
+                Map<String, String> contributionVariables = getEnvVarsByContribution(build, this, logger, listener);
+
+                final Map<String, String> mergedVariables = envInjectEnvVarsService.getMergedVariables(
+                        infraEnvVarsNode,
+                        propertiesVariables,
+                        groovyMapEnvVars,
+                        contributionVariables);
+
+                //Add an action to share injected environment variables
+                new EnvInjectActionSetter(rootPath).addEnvVarsToEnvInjectBuildAction(build, mergedVariables);
+            }
+        } catch (Throwable t) {
+            StringWriter sw = new StringWriter();
+            PrintWriter pw = new PrintWriter(sw);
+            t.printStackTrace(pw);
+            logger.error("Failure while executing scripts in a prebuild phase: " + sw.toString());
+            pw.close();
+            return false;
+        }
+        return true;
+    }
+
+    private Node getNode() {
+        Computer computer = Computer.currentComputer();
+        if (computer == null) {
+            return null;
+        }
+        return computer.getNode();
+    }
+
+    private FilePath getNodeRootPath() {
+        Node node = getNode();
+        if (node != null) {
+            return node.getRootPath();
+        }
+        return null;
+    }
+    
+    private Map<String, String> getEnvVarsByContribution(AbstractBuild build, EnvInjectPrebuildJobProperty envInjectJobProperty,
+            EnvInjectLogger logger, BuildListener listener) throws EnvInjectException {
+
+        assert envInjectJobProperty != null;
+        Map<String, String> contributionVariables = new HashMap<String, String>();
+
+        EnvInjectJobPropertyContributor[] contributors = envInjectJobProperty.getContributors();
+        if (contributors != null) {
+            logger.info("Injecting contributions.");
+            for (EnvInjectJobPropertyContributor contributor : contributors) {
+                contributionVariables.putAll(contributor.getEnvVars(build, listener));
+            }
+        }
+        return contributionVariables;
+    }
+
+    
+    @Deprecated
+    private transient boolean injectGlobalPasswords;
```

Comment:
@daniel-beck I've turned on this like an experiment, sorry for spam. (Already turned off). 

I'll create very light profile for jenkins plugins to avoid spam in future, as of it too many violations of code style in any place :(